robot/robot.py

```diff
@@ -58,7 +58,12 @@ def _update_boards(self, known_boards, board_type, directory_name):
         # Add all boards that weren't previously there
         for board in new_paths - known_paths:
             print('New board found:', board)
-            boards.append(board_type(board))
+
+            try:
+                new_board = board_type(board)
+                boards.append(new_board)
+            except ConnectionRefusedError as e:
```

Comment:
`FileNotFoundError` I would agree, the others shouldn't happen during a call to `socket.connect`.